pkg/diagnostics/metrics_test.go

```diff
@@ -9,74 +14,145 @@ import (
+# TYPE process_resident_memory_bytes gauge
+process_resident_memory_bytes 1.644544e+07
+# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
+# TYPE process_start_time_seconds gauge
+process_start_time_seconds 1.5813370923e+09
+# HELP process_virtual_memory_bytes Virtual memory size in bytes.
+# TYPE process_virtual_memory_bytes gauge
+process_virtual_memory_bytes 6.53000704e+08
+# HELP process_virtual_memory_max_bytes Maximum amount of virtual memory available in bytes.
+# TYPE process_virtual_memory_max_bytes gauge
+process_virtual_memory_max_bytes -1
+# HELP promhttp_metric_handler_requests_in_flight Current number of scrapes being served.
+# TYPE promhttp_metric_handler_requests_in_flight gauge
+promhttp_metric_handler_requests_in_flight 1
+# HELP promhttp_metric_handler_requests_total Total number of scrapes by HTTP status code.
+# TYPE promhttp_metric_handler_requests_total counter
+promhttp_metric_handler_requests_total{code="200"} 0
+promhttp_metric_handler_requests_total{code="500"} 0
+promhttp_metric_handler_requests_total{code="503"} 0`
+)
 
 func TestBaselineMetrics(t *testing.T) {
 	w := httptest.NewRecorder()
 	req, _ := http.NewRequest("GET", "/", nil)
 	promhttp.Handler().ServeHTTP(w, req)
-	assertMetricsExist(t, w.Body.String(), testdata.BaselineMetrics)
-}
-
-func TestMetricsGRPCMiddlewareStream(t *testing.T) {
-	opt := grpc_go.StreamInterceptor(MetricsGRPCMiddlewareStream())
-	s := grpc_go.NewServer(opt)
-	pb.RegisterGreeterServer(s, &server{})
-	grpc_prometheus.Register(s)
-
-	w := httptest.NewRecorder()
-	req, _ := http.NewRequest("GET", "/", nil)
-	promhttp.Handler().ServeHTTP(w, req)
-	assertMetricsExist(t, w.Body.String(), testdata.GRPCStreamMetrics)
-}
-
-func TestMetricsGRPCMiddlewareUnary(t *testing.T) {
-	opt := grpc_go.UnaryInterceptor(MetricsGRPCMiddlewareUnary())
-	s := grpc_go.NewServer(opt)
-	pb.RegisterGreeterServer(s, &server{})
-	grpc_prometheus.Register(s)
-
-	w := httptest.NewRecorder()
-	req, _ := http.NewRequest("GET", "/", nil)
-	promhttp.Handler().ServeHTTP(w, req)
-	assertMetricsExist(t, w.Body.String(), testdata.GRPCUnaryMetrics)
-}
-
-func TestMetricsHTTPMiddleware(t *testing.T) {
-	fakeLogger := logger.NewLogger("test")
-	inner := func(ctx *fasthttp.RequestCtx) {}
-	mw := MetricsHTTPMiddleware(inner)
-	outter := nethttpadaptor.NewNetHTTPHandlerFunc(fakeLogger, mw)
-
-	// HTTP metrics are not initialized with a zero value
-	// so we have to invoke the metrics middleware manually.
-	mwResWriter := httptest.NewRecorder()
-	mwReq, _ := http.NewRequest("GET", "/", strings.NewReader(""))
-	outter.ServeHTTP(mwResWriter, mwReq)
-
-	metricsResWriter := httptest.NewRecorder()
-	metricsReq, _ := http.NewRequest("GET", "/", strings.NewReader(""))
-	promhttp.Handler().ServeHTTP(metricsResWriter, metricsReq)
-
-	assertMetricsExist(t, metricsResWriter.Body.String(), testdata.HTTPServerMetrics)
+	assertMetricsExist(t, w.Body.String(), baselineMetrics)
```

Comment:
I was tracking an [issue](https://github.com/dapr/dapr/issues/1105) to change the way we parse the metrics to use `expfmt`. There's an example of how to do this in the [e2e metrics test](https://github.com/dapr/dapr/blob/e1c8b4ee2cbb3705cba40165493cd201ccf68f4f/tests/e2e/metrics/metrics_test.go#L140). I think it probably makes sense to close this out with this change too.

Reply:
Yeah your e2e metrics tests are more valuable to validate if the metrics is up and flowing rather than adding unit-test.

Reply:
Also I realized that we're testing prometheus http pkg :) so it makes more sense to move this unit-test to your e2e tests.